frontend/src/pages/search.vue

```diff
@@ -15,146 +15,145 @@
-        (item: BaseItemDto) => item.Type === 'Book'
-      );
-
-      this.personSearchResults = (
-        await this.$remote.sdk.newUserApi(getPersonsApi).getPersons({
-          userId: this.$remote.auth.currentUserId.value,
-          searchTerm: this.searchQuery
-        })
-      ).data.Items;
-
-      this.artistSearchResults = itemResults.filter(
-        (item: BaseItemDto) => item.Type === 'MusicArtist'
-      );
-
-      this.loading = false;
+import { useRemote, useResponsiveClasses } from '@/composables';
+
+const route = useRoute();
+const router = useRouter();
+const remote = useRemote();
+
+const loading = ref(false);
+const searchTab = ref(0);
+const topSearchResults = ref<BaseItemDto[]>([]);
+const movieSearchResults = ref<BaseItemDto[]>([]);
+const showSearchResults = ref<BaseItemDto[]>([]);
+const albumSearchResults = ref<BaseItemDto[]>([]);
+const trackSearchResults = ref<BaseItemDto[]>([]);
+const bookSearchResults = ref<BaseItemDto[]>([]);
+const personSearchResults = ref<BaseItemDto[]>([]);
+const artistSearchResults = ref<BaseItemDto[]>([]);
+
+const searchQuery = computed(() => {
+  return route.query?.q?.toString() || '';
+});
+
+watch(
+  () => searchQuery,
+  (newQuery) => {
+    if (newQuery.value !== '') {
+      performSearchDebounce();
     }
   }
+);
+
+onBeforeMount(async () => {
+  if (searchQuery.value === '') {
+    router.back();
+  } else {
+    await performSearch();
+  }
 });
+
+const performSearchDebounce = debounce(async () => await performSearch(), 500);
+
+/**
+ * Search in Jellyfin
+ */
+async function performSearch(): Promise<void> {
+  loading.value = true;
+
+  const itemResults = (
+    await remote.sdk.newUserApi(getItemsApi).getItemsByUserId({
+      userId: remote.auth.currentUserId.value || '',
+      searchTerm: searchQuery.value,
+      includeItemTypes: [
+        BaseItemKind.Movie,
+        BaseItemKind.Series,
+        BaseItemKind.Audio,
+        BaseItemKind.MusicAlbum,
+        BaseItemKind.Book,
+        BaseItemKind.MusicArtist,
+        BaseItemKind.Person
+      ],
+      recursive: true
+    })
+  ).data.Items;
+
+  if (itemResults) {
+    topSearchResults.value = itemResults.slice(0, 24);
+
+    movieSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Movie'
+    );
+
+    showSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Series'
+    );
+
+    albumSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'MusicAlbum'
+    );
+
+    trackSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Audio'
+    );
+
+    bookSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Book'
+    );
+
+    personSearchResults.value = (
+      await remote.sdk.newUserApi(getPersonsApi).getPersons({
+        userId: remote.auth.currentUserId.value,
+        searchTerm: searchQuery.value
+      })
+    ).data.Items;
```

Comment:
```suggestion
    ).data.Items || [];
```
Should do the trick :)